api/handlers.go

```diff
@@ -1225,6 +1242,25 @@ func (api *API) checkUserQuotas(ctx context.Context, u *database.User) {
 	}
 }
 
+// userFromRequest checks the requests for various forms of authentication (API
+// key, cookie, authorization header) and returns user information based on
+// those.
+func (api *API) userFromRequest(req *http.Request) (*database.User, jwt2.Token, error) {
+	// Check for an API key.
+	u, tk, err := api.userAndTokenByAPIKey(req)
+	if err != nil && !errors.Contains(err, ErrNoAPIKey) {
```

Comment:
we're changing the error handling here quite a bit, right? before we were checking all kinds of errors and sometimes even return an internal server error, it seems like that's all gone now, should we consider this is a breaking change wrt backwards compat? 

I don't think so in this case, but I just wanted to comment on it because I'm not sure why all of those error checks are not needed anymore all of a sudden

Reply:
The only change of behaviour is that `internal server errors` coming from processing API keys are now replaced by `unauthorized` errors. This has always been the standard behaviour when processing cookies and authorization tokens. Having in mind that API keys haven't really been announced, I wouldn't worry about being that strict with backwards compatibility.

If we want to be stricter with this and give the user really accurate information why their request is being denied, I can go over all the possible errors that can be returned by `userAndTokenByAPIKey` and `userAndTokenByRequestToken` and add proper responses.

Reply:
ok, I was talking about `database.ErrUserNotFound` in particular, we used to handle that explicitly and now we don't anymore, but if you say it's fine then I'm fine with that